tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -177,142 +199,180 @@ public int PartialFlushMinSpans
         /// </summary>
         internal MetricsTransportType MetricsTransport { get; set; }
 
-        private void ConfigureMetricsTransport(IConfigurationSource source, bool forceMetricsOverUdp)
-        {
-            MetricsTransportType? metricsTransport = null;
-
-            var dogStatsdPort = source?.GetInt32(ConfigurationKeys.DogStatsdPort);
+        internal List<string> ValidationWarnings { get; }
 
-            MetricsPipeName = source?.GetString(ConfigurationKeys.MetricsPipeName);
+        private void ConfigureMetricsTransport(string agentHost, int dogStatsdPort, string metricsPipeName, string metricsUnixDomainSocketPath)
+        {
+            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
+            // The agent will fail to start if it can not bind a port, so we need to override 8126 to prevent port conflict
+            // Port 0 means it will pick some random available port
+            if (dogStatsdPort < 0)
+            {
+                ValidationWarnings.Add("The provided dogStatsD port isn't valid, it should be positive.");
+            }
 
-            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict.
-            // The agent will fail to start if it can not bind a port.
-            // If the dogstatsd port isn't explicitly configured, check for pipes or sockets.
-            if (!forceMetricsOverUdp && (dogStatsdPort == 0 || dogStatsdPort == null))
+            if (dogStatsdPort > 0 || agentHost != null)
```

Comment:
We've lost the `forceMetricsOverUdp` check here. I _think_ that's right, because it looks like that check was just based on whether `agentHost` was explictly provided, but just double checking 🙂

Reply:
Good catch. There was a bug as Colin used to have the agent URL in the agentHost before. So the result is the same but the order of precedence was lost. 
Fixed in e374810ab059e20439f580cf1f4d046a17784a70

Reply:
Hmmm... but now ConfigureMetricsSettings is dependent on AgentURI, so we probably need to update metrics too when users set the agent uri? 🤔